analyses/cell-type-wilms-tumor-06/scripts/prepare-fetal-references.R

```diff
@@ -0,0 +1,215 @@
+  refdata <- lapply(X = annotation_levels, function(x) {
+    reference[[x, drop = TRUE]]
+  })
+  names(refdata) <- annotation_levels
+
+  # Create and return list of reference components
+  reference_list <- list(
+    reference = reference,
+    dims = dims,
+    annotation_levels = annotation_levels,
+    refdata = refdata
+  )
+
+  return(reference_list)
+}
+
+# Define paths -------------------------------------------
+
+project_root <- rprojroot::find_root(rprojroot::is_git_root)
+module_dir <- file.path(
+  project_root,
+  "analyses",
+  "cell-type-wilms-tumor-06"
+)
+
+
+# Create output directory and define paths for final reference files
+output_dir <- file.path(
+  module_dir,
+  opts$output_dir
+)
+dir.create(output_dir, recursive = TRUE, showWarnings = FALSE)
+stewart_ref_file <- file.path(
+  output_dir,
+  "stewart_formatted_ref.rds"
+)
+cao_ref_file <- file.path(
+  output_dir,
+  "cao_formatted_ref.rds"
+)
+homologs_file <- file.path(
+  output_dir,
+  "homologs.rds"
+)
+
+# Define the annotation levels to use for each reference ----------------
+stewart_annotation_levels <- c("compartment", "cell_type")
+cao_annotation_levels <- c("annotation.l1", "annotation.l2", "organ")
+
+
+# Prepare Stewart (fetal kidney) reference ------------------------------
+
+# path where downloaded RDS file will be stored
+path_to_data <- file.path(
+  module_dir,
+  "scratch",
+  "fetal_kidney.rds"
+)
+
+# Download data
+download.file(url = opts$kidney_url, destfile = path_to_data)
+
+# Read in data
+seurat <- readRDS(path_to_data)
+
+# Transform and dimension reduction
+set.seed(opts$seed)
+options(future.globals.maxSize = 891289600000)
+
+s <- SCTransform(
+  seurat,
+  verbose = FALSE,
+  method = "glmGamPoi",
+  conserve.memory = TRUE
+)
+s <- RunPCA(s, npcs = 50, verbose = FALSE)
+s <- RunUMAP(s, dims = 1:50, verbose = FALSE, return.model = TRUE)
+
+# Create reference
+fetal_kidney <- AzimuthReference(
+  s,
+  refUMAP = "umap",
+  refDR = "pca",
+  refAssay = "SCT",
+  dims = 1:50,
+  k.param = 31,
+  plotref = "umap",
+  plot.metadata = NULL,
+  ori.index = NULL,
+  colormap = NULL,
+  assays = NULL,
+  metadata = stewart_annotation_levels,
+  reference.version = "0.0.0",
+  verbose = FALSE
+)
+
+# format for label transfer
+stewart_ref_list <- prepare_azimuth_reference(fetal_kidney, stewart_annotation_levels)
+
+# export formatted reference
+saveRDS(stewart_ref_list, stewart_ref_file)
+
+
+
+# Prepare Cao (full fetal organ) reference ------------------------------
+
+# Load in the reference, keeping only the $map portion
+fetus_ref <- SeuratData::LoadData("fetusref", type = "azimuth")$map
+
+
+# format for label transfer
+cao_ref_list <- prepare_azimuth_reference(fetus_ref, cao_annotation_levels)
+
+# export formatted reference
+saveRDS(cao_ref_list, cao_ref_file)
+
+
+# Finally, download the ID mapping file from Seurat ---------------------
+download.file(url = opts$homologs_url, destfile = homologs_file)
```

Comment:
Without knowing exactly how this will be used, I'm not sure we want to do this? We already have all of the relevant mappings stored within the SCE objects, and I don't think we know what versions this particular file is/will be based on, so it might well be mismatched. If we do need this particular structure, we can generate it?

Also, for transparency and easier later updates, I would not perform this step within this script, but as an external step.

Reply:
We do need this particular structure - this is specifically adapted from Azimuth's code which downloads/uses this file directly. But, indeed, no idea how versioning will play out, and..yeah.

We could generate something like this on our own for mapping, though:
```
> head(h)
  Gene.stable.ID.human Transcript.stable.ID.human Gene.name.human
1      ENSG00000204479            ENST00000376098        PRAMEF17
2      ENSG00000204479            ENST00000376098        PRAMEF17
3      ENSG00000204479            ENST00000376098        PRAMEF17
4      ENSG00000204479            ENST00000376098        PRAMEF17
5      ENSG00000204479            ENST00000376098        PRAMEF17
6      ENSG00000204479            ENST00000376098        PRAMEF17
  Transcript.name.human Gene.stable.ID.mouse Transcript.stable.ID.mouse
1          PRAMEF17-201   ENSMUSG00000094741         ENSMUST00000178291
2          PRAMEF17-201   ENSMUSG00000074720         ENSMUST00000099245
3          PRAMEF17-201   ENSMUSG00000094337         ENSMUST00000179140
4          PRAMEF17-201   ENSMUSG00000094337         ENSMUST00000164489
5          PRAMEF17-201   ENSMUSG00000095570         ENSMUST00000179972
6          PRAMEF17-201   ENSMUSG00000096728         ENSMUST00000178296
  Gene.name.mouse Transcript.name.mouse
1      AC140325.3        AC140325.3-201
2      AC140325.1        AC140325.1-201
3      AC140325.2        AC140325.2-201
4      AC140325.2        AC140325.2-202
5      AC140325.4        AC140325.4-201
6      AC140325.5        AC140325.5-201
```

(FWIW, I have all the code written on my end for label transfer working on my end, and it's slowing debuting in PRs 😄 )

Reply:
We can wait to resolve this until the file is being used. I feel like we should be able to bypass Azimuth's conversion by converting the feature names from `gene_id` to `gene_symbol` using the `rowData` before/during conversion to Seurat objects, which will allow us to be explicit and consistent.

Reply:
Noting from separate discussion that this is definitely a "nice to have" for the module, but not an urgent change to do now since we have some deadlines approaching. We may circle back to this in the future!